notebooks/Sigma_test_Notebook.ipynb

```diff
@@ -0,0 +1,340 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rule1.es_query"
+   ]
+  },
+  {
+   "source": [
+    "# Single rule"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Single Rule tests\n",
+    "single_rule = ts_client.get_sigma_rule(rule_uuid='5266a592-b793-11ea-b3de-0242ac130004')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "single_rule.__dict__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "single_rule.id"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "single_rule.rule_uuid\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "single_rule.es_query"
+   ]
+  },
+  {
+   "source": [
+    "# Text Sigma examples"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "MOCK_SIGMA_RULE = \"\"\"\n",
+    "title: Suspicious Installation of Zenmap\n",
+    "id: 5266a592-b793-11ea-b3de-0242ac130004\n",
+    "description: Detects suspicious installation of Zenmap\n",
+    "references:\n",
+    "    - https://rmusser.net/docs/ATT&CK-Stuff/ATT&CK/Discovery.html\n",
+    "author: Alexander Jaeger\n",
+    "date: 2020/06/26\n",
+    "modified: 2021/01/01\n",
+    "logsource:\n",
+    "    product: linux\n",
+    "    service: shell\n",
+    "detection:\n",
+    "    keywords:\n",
+    "        # Generic suspicious commands\n",
+    "        - '*apt-get install zmap*'\n",
+    "    condition: keywords\n",
+    "falsepositives:\n",
+    "    - Unknown\n",
+    "level: high\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rule_by_text = ts_client.get_sigma_rule_by_text(MOCK_SIGMA_RULE)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rule_by_text.es_query"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rule_by_text.references"
+   ]
+  },
+  {
+   "source": [
+    "# Analyzer"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "timelines = sketch.list_timelines()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "timelines[0].name #sigma_events.jsonl"
```

Comment:
well... why is that the sigma_events timeline? what if you had added other timeline there before?

you could do something like

```python
timeline = None
for timeline_ in timelines:
  if timeline_.name == 'sigma_events.jsonl':
    timeline = timeline_
    break
```

and something like that, then you could have a guarantee....

Reply:
ack